### PR TITLE
Solution clean-up and fixing integration tests

### PR DESCRIPTION
**Pull request by SpectacledBear**
I had two goals with this pull request: To clean up the solution in general and to get the integration tests passing again.

For the cleanup, I removed the incomplete tools directory, implemented NuGet Package Restore and added the missing integration test project configuration file.

For fixing the integration tests, I split the tests in several classes, added and organized utility classes into directories and tidied up the HttpWebResponse disposal issues that were causing multiple tests to fail.

### CODE REVIEW

**Comment by SpectacledBear** [comment]
Hey @j3parker and @davidcollie, here is the first of two PRs I'd like to submit for the C# Valence SDK.

File: lib/D2L.Extensibility.AuthSdk.IntegrationTests/AnonymousRequestTests.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		private HttpWebRequest m_request;
+
+		[SetUp]
+		public void TestSetup() {
+			m_request = RequestProvider.PrepareApiRequest( ContextProvider.AnonUserContext(), RouteProvider.VersionsRoute );
+		}
+
+		[Test]
+		public void SendAnonymousRequest_ResponseReceived() {
+			Assert.DoesNotThrow( () => { using( HttpWebResponse response = m_request.GetResponse() as HttpWebResponse ) { } } );
```
Comment (j3parker) [inline lib/D2L.Extensibility.AuthSdk.IntegrationTests/AnonymousRequestTests.cs]: (nit) whitespace here could be better

**Comment by omsmith** [comment]
Lots of changes so hard to actually reason about, but given it's mostly tests and whatnot, seems alright to me.

**Comment by j3parker** [comment]
Yeah - LGTM. Kyle and I talked in person.

**Comment by omsmith** [comment]
I don't have the merge button, myself

File: lib/D2L.Extensibility.AuthSdk.IntegrationTests/AnonymousRequestTests.cs (hunk 1 above)
Comment (SpectacledBear) [inline lib/D2L.Extensibility.AuthSdk.IntegrationTests/AnonymousRequestTests.cs]: I simplified the using statement and hopefully improved the whitespace a bit with commit 338c23e.

**Pull request merged**